src/ORM/AssociationCollection.php

```diff
@@ -85,6 +85,22 @@ public function getByProperty($prop)
     }
 
     /**
+     * Fetch an association by foreign key.
+     *
+     * @param string $fk The foreign key to find an association by.
+     * @return \Cake\ORM\Association|null Either the association or null.
+     */
+    public function getByForeignKey($fk)
```

Comment:
What if a foreign key is used for multiple associations? For example:

* Articles hasMany Comments (just a foreign key)
* Articles hasOne TopComment (foreign key, plus conditions)

Wouldn't this function be less useful?